template/src/features/map/map.hooks.ts

```diff
@@ -0,0 +1,74 @@
+import { useSelector, useDispatch } from 'react-redux'
+import { useCallback, useEffect, useState } from 'react'
+import { ViewportProps } from 'react-map-gl'
+import { selectViewport } from 'routes/routes.selectors'
+import { updateQueryParams } from 'routes/routes.actions'
+import useDebounce from 'hooks/debounce.hooks'
+import {
+  selectGeneratorsConfig,
+  updateGenerator,
+  MapCoordinates,
+  UpdateGeneratorPayload,
+} from './map.slice'
+
+export const useGeneratorsConnect = () => {
+  const dispatch = useDispatch()
+  return {
+    globalConfig: useSelector(selectViewport),
+    generatorsConfig: useSelector(selectGeneratorsConfig),
+    updateGenerator: (payload: UpdateGeneratorPayload) => dispatch(updateGenerator(payload)),
+  }
+}
+
+type UseViewport = {
+  viewport: MapCoordinates
+  onViewportChange: (viewport: ViewportProps) => void
+  setMapCoordinates: (viewport: ViewportProps) => void
+}
+export function useDebouncedViewport(
```

Comment:
This will depend if it is needed by the app performance, anyone without router won't be necessary, so I agree on having the generic useDebounce but I think doesn't worth to move this one